### PR TITLE
Saving interactions to JSON

### LINKED ISSUE #75: Alllow saving of interactions in JSON

### PR DESCRIPTION
**Pull request by MagnificentCreature**
Changes:
Interactions can now be saved as JSON (#75)
Added test cases to check this functionality
Small bug fixes when creating interaction without outcomes
Created an abstraction for a list of interactions

### CODE REVIEW

File: build.gradle
Comment (KamiliArsyad) [inline build.gradle:1]: Not sure about this

File: src/main/java/seedu/address/model/person/interaction/InteractionList.java
Comment (KamiliArsyad) [inline src/main/java/seedu/address/model/person/interaction/InteractionList.java:1]: Semantically the method names should be made clearer. When getting an Interaction, am I getting the interaction, the interaction list, or an array list of interactions? I think it helps getting these clear purely from the method names.

Comment (KamiliArsyad) [reply src/main/java/seedu/address/model/person/interaction/InteractionList.java:1]: I think for me the confusing part is mainly how InteractionList doesn't really abstract away ArrayList<Interaction>. This won't affect anything directly as of right now but might want to change in the future as this results in a lot of redundancy

File: src/main/java/seedu/address/model/person/interaction/InteractionList.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+     * @param index The index of the {@code Interaction} to return.
+     */
+    public Interaction getInteraction(int index) {
+        return this.interactions.get(index);
+    }
+
+    /**
+     * Returns the ArrayList of {@code Interaction}.
+     */
+    public ArrayList<Interaction> getInteractions() {
```
Comment (KamiliArsyad) [inline src/main/java/seedu/address/model/person/interaction/InteractionList.java]: `getInteractionArrayList` (?)

File: src/main/java/seedu/address/model/person/interaction/InteractionList.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        return this.interactions;
+    }
+
+    /**
+     * Returns a filtered list of {@code Interaction} that matches the given predicate.
+     */
+    public InteractionList getFilteredInteraction(Predicate<Interaction> predicate) {
```
Comment (KamiliArsyad) [inline src/main/java/seedu/address/model/person/interaction/InteractionList.java]: `getFilteredInteractionList` or `getFilteredInteractions`?

**Review by KamiliArsyad — APPROVED** [review]

File: build.gradle
Comment (MagnificentCreature) [reply build.gradle:1]: Oh this wont affect you at all

File: src/main/java/seedu/address/model/person/interaction/InteractionList.java
Comment (MagnificentCreature) [reply src/main/java/seedu/address/model/person/interaction/InteractionList.java:1]: I'll change the method names. 
The abstraction might be useful in the future when we add more filtering methods and things that have to do with the reminders?

**Pull request merged**